.github/workflows/promote-canary-to-latest.yml

```
@@ -0,0 +1,29 @@
+# Workflow to manually promote the returntocorp/semgrep:canary docker image
+# to returntocorp/semgrep:latest that most customers are using
+
+name: Promote returntocorp/semgrep:canary to :latest
+
+on:
+  workflow_dispatch:
+    inputs:
+      confirmed:
+        description: "Are you sure you want to do this? This is a sensitive operation"
+        type: boolean
+        required: true
+        default: false
+
+jobs:
+  promote:
+    name: promote :canary to :latest
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - run: |
+          if [[ "${{ inputs.confirmeed }}" == "true" ]]; then
+             docker pull returntocorp/semgrep:canary
+             docker tag  returntocorp/semgrep:canary returntocorp/semgrep:latest
+             docker push returntocorp/semgrep:latest
+          fi
```

I'd consider approaching the input like this 
```suggestion      
          docker pull returntocorp/semgrep:canary
          docker tag  returntocorp/semgrep:canary returntocorp/semgrep:latest
          if [[ "${{ inputs.confirmeed }}" == "true" ]]; then
             docker push returntocorp/semgrep:latest
          else
             echo "dry-run: not pushing tag upstream
          fi
```